folium/folium.py

```diff
@@ -251,7 +252,10 @@ def choropleth(self, geo_path=None, geo_str=None, data_out='data.json',
 
         # Create GeoJson object
         if geo_path:
-            geo_data = open(geo_path)
+            if geo_path.lower().startswith(('http', 'ftp', 'https')):
```

Comment:
Probably should include the colon, too; someone could have a file that starts with any of these sequences.